### PR TITLE
docs: how to customize the search experience

### PR DESCRIPTION
**Pull request by camelliazhang**
Initial documentation with high level instructions, will probably follow up with a concrete example. 

## Checklist
- [x] The PR conforms to DataHub's [Contributing Guideline](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [x] Docs related to the changes have been added/updated (if applicable)

### CODE REVIEW

File: docs/how/customize-elasticsearch-query-template.md [hunk 1]
```diff
@@ -0,0 +1,103 @@
+# How to customize the search experience of an entity?
+
+> Assume you have all the fields needed for your query, otherwise, refer to [this doc](./search-over-new-field.md) if you're only interested in indexing a new field of an existing entity 
+
+## 1. Default query template for each type of search document
+The search query is constructed and executed through [Search DAO]. Each search document is associated with a customizable query template, which can found [here](https://github.com/linkedin/datahub/tree/master/gms/impl/src/main/resources). 
+Take [dataset search query template] for example, it supports a few features:
+
+The general search accepts a query and at run time, [ES Search DAO] will replace $INPUT in the query template and generate the query for Elastic Search.
```
Comment (mars-lan) [inline docs/how/customize-elasticsearch-query-template.md:9]: s/Elastic Search/Elasticsearch/g

File: docs/how/customize-elasticsearch-query-template.md [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+       "filter": {
+         "term": {
+           "deprecated": true
+         }
+       },
+       "weight": 0.5
+     }, ...
+```
+
+## 2. Elastic Search Full Text Queries
```
Comment (mars-lan) [inline docs/how/customize-elasticsearch-query-template.md:49]: Use title case for all titles?

File: docs/how/customize-elasticsearch-query-template.md (hunk 1 above)
Comment (camelliazhang) [reply docs/how/customize-elasticsearch-query-template.md]: Thanks, fixed the entire doc to keep consistent.

File: docs/how/customize-elasticsearch-query-template.md (hunk 2 above)
Comment (camelliazhang) [reply docs/how/customize-elasticsearch-query-template.md]: Fixed

**Pull request merged**